frontend/packages/noobaa-storage-plugin/src/components/bucket-class/backingstore-table.tsx

```diff
@@ -0,0 +1,247 @@
+import * as React from 'react';
+import * as classNames from 'classnames';
+import * as _ from 'lodash';
+import { compose } from 'redux';
+import { Title, Flex, FlexItem, Button, FormGroup, Form, pluralize } from '@patternfly/react-core';
+import { PlusCircleIcon } from '@patternfly/react-icons';
+import { IRow } from '@patternfly/react-table';
+import { getNamespace, getName, useSelectList, getUID } from '@console/shared';
+import { Table, ListPage, getFilteredRows } from '@console/internal/components/factory';
+import { ResourceLink } from '@console/internal/components/utils';
+import { referenceForModel } from '@console/internal/module/k8s';
+import { getBucketName, getRegion, getBackingStoreType, getBSLabel } from '../../utils';
+import CreateBackingStoreFormModal from '../create-backingstore-page/create-bs-modal';
+import { NooBaaBackingStoreModel } from '../../models';
+import { BackingStoreKind, PlacementPolicy } from '../../types';
+import './_backingstore-table.scss';
+
+const tableColumnClasses = [
+  '',
+  classNames('pf-m-hidden', 'pf-m-visible-on-sm'),
+  classNames('pf-m-hidden', 'pf-m-visible-on-sm'),
+  classNames('pf-m-hidden', 'pf-m-visible-on-sm'),
+];
+
+const getColumns = () => [
+  {
+    title: 'Name',
+    props: { className: tableColumnClasses[0] },
+  },
+  {
+    title: 'Bucket Name',
+    props: { className: tableColumnClasses[1] },
+  },
+  {
+    title: 'Type',
+    props: { className: tableColumnClasses[2] },
+  },
+  {
+    title: 'Region',
+    props: { className: tableColumnClasses[3] },
+  },
+];
+
+const getRows: GetRows = (rowProps, selectedItems) => {
+  const {
+    componentProps: { data },
+  } = rowProps;
+
+  const rows = data.map((bs) => {
+    const cells: IRow['cells'] = [
+      {
+        title: (
+          <ResourceLink
+            kind={referenceForModel(NooBaaBackingStoreModel)}
+            name={getName(bs)}
+            namespace={getNamespace(bs)}
```

Comment:
You also want to handle to close the modal, when any resource is clicked here.
Otherwise, you will have the modal opened.